src/session_concurrency/payload_dataset_manager.cc

```diff
@@ -85,6 +85,12 @@ PayloadDatasetManager::GetDelayForPayload(size_t dataset_index) const
     throw std::runtime_error(error.Message());
   }
 
+  if (!delay_ms_tensor_data.data_ptr) {
+    throw std::runtime_error(
+        "Missing 'delay' input for payload with index " +
+        std::to_string(dataset_index));
+  }
+
```

Comment:
Good catch @dyastremsky , the code didn't fully check for the missing delay value case, and now it does, and has been tested both in unit tests and manually :)

Reply:
Nice job, Matt!